django_filters/filters.py

```diff
@@ -484,6 +486,38 @@ class TimeRangeFilter(RangeFilter):
     field_class = TimeRangeField
 
 
+class SearchVectorFilter(Filter):
+
```

Comment:
Instead of overloading the purpose of `field_name`, create a custom `__init__` that requires users to pass in a list of search fields. eg, 

```python
# Note that this provides a default `field_name` to be used for the annotation
# and query, and that `search_fields` will be a required keyword argument.
def __init__(self, field_name='search_vector', lookup_expr='exact', *, search_fields, **kwargs):
    super().__init__(field_name, lookup_expr, **kwargs)

    self.search_fields = search_fields
```

From here, it's really simple to overload the `.filter()` method to add in the annotation

```python
def filter(self, qs, value):
    # Note that the annotation is operation on the filter's `field_name`, which was
    # set above to 'search_vector' as the filter's default value.
    qs.annotate(self.field_name=SearchVector(self.search_fields)
    
    return super().filter(qs, value)
```

Reply:
Yes, this was the way it should be implemented. Thanks.